json_serialization/reader.nim

```diff
@@ -180,9 +180,10 @@ func maxAbsValue(T: type[SomeInteger]): uint64 {.compileTime.} =
   else: uint64(high(T))
 
 proc parseJsonNode(r: var JsonReader): JsonNode
-                  {.gcsafe, raises: [IOError, JsonReaderError, Defect].}
+                  {.gcsafe, raises: [UnexpectedValueError, IOError, JsonReaderError, Defect].}
```

Comment:
this looks like a regression in nim 1.7 - ie raising new exceptions is a breaking change and should be reported upstream so that it can be undone at the source

Reply:
actually, nevermind - this is an issue in json-ser indeed and the change is correct - if anything, earlier nim versions are buggy for not detecting it

Reply:
No, this change is not correct and this is indeed a Nim bug. `UnexpectedValueError` is a sub-type of `JsonReaderError`, so the signature was already covering it.